packages/noya-state/src/reducers/application.ts

```diff
@@ -875,6 +882,77 @@ export function reducer(
         state.selectedSwatchIds = [swatch.do_objectID];
       });
     }
+    case 'duplicateSwatch': {
+      const [, id] = action;
+      const ids = typeof id === 'string' ? [id] : id;
+
+      return produce(state, (state) => {
+        const sharedSwatches = state.sketch.document.sharedSwatches ?? {
+          _class: 'swatchContainer',
+          do_objectID: uuid(),
+          objects: [],
+        };
+
+        sharedSwatches.objects.forEach((swatch: Sketch.Swatch) => {
+          if (!ids.includes(swatch.do_objectID)) return;
+          const swatchColor = swatch.value;
+
+          const lastName = sharedSwatches.objects
```

Comment:
This will work if the names are in the correct order, but if the user rearranges the swatches, there may be a larger number that comes first: `["Test 3", "Test 2", "Test"]`. Since I'm hoping to use this same function for layers in the future, I think we want to handle that case.

Let's update `getIncrementedName` to also take the array of other names. Maybe something like:

```ts
function getIncrementedName(originalName: string, names: string[] = []): string {
  // ...
  const numbers = [originalName, ...names]
    .filter(name => name.startsWith(prefix))
    .map(name => /* map to just the parsed number */)
    .sort()
  const nextNumber = numbers.length > 0 ? numbers[numbers.length - 1] : 1
  // ...
}
```

Then to use it here:

```ts
const names = sharedSwatches.objects.map(s => s.name)
const newName = getIncrementedName(swatch.name, names)
```

Reply:
yes I was also thinking this could be handle in the same function!